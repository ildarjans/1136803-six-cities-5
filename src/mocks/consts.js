export const MOCK_OFFERS_COUNT = 100;
export const MOCK_REVIEWS_COUNT = 5;
export const AVATAR_URL = `https://avatars.dicebear.com/api`;
export const MAX_FEATURES = 3;
export const imageGallery = {
  MIN: 2,
  MAX: 8,
};

export const priceRange = {
  MIN: 19,
  MAX: 4999,
};
export const bedroomsRange = {
  MIN: 1,
  MAX: 6,
};
export const guestRange = {
  MIN: 1,
  MAX: 10,
};

export const avatarSprites = [
  `male`,
  `female`,
  `human`,
  `bottts`,
  `avataaars`,
  `gridy`,
];


export const names = [
  `James`,
  `John`,
  `Robert`,
  `William`,
  `Joseph`,
  `Charles`,
  `Thomas`,
  `Christopher`,
  `Anthony`,
  `Mark`,
  `Paul`,
  `Steven`,
  `Kenneth`,
  `Edward`,
  `Kevin`,
  `Ronald`,
  `Jason`,
  `Jeffrey`,
  `Gary`,
  `Nicholas`,
  `Stephen`,
  `Jonathan`,
  `Frank`,
  `Justin`,
  `Gregory`,
  `Samuel`,
  `Mary`,
  `Patricia`,
  `Elizabeth`,
  `Jessica`,
  `Sarah`,
  `Dorothy`,
  `Karen`,
  `Nancy`,
  `Sandra`,
  `Jennifer`,
  `Amanda`,
  `Rebecca`,
  `Sharon`,
  `Cynthia`,
  `Pamela`,
  `Katherine`,
  `Nicole`,
  `Samantha`,
  `Carolyn`,
];
export const lastNames = [
  `Smith`,
  `Brown`,
  `Johnson`,
  `Jones`,
  `Williams`,
  `Davis`,
  `Clark`,
  `White`,
  `Moore`,
  `Thompson`,
  `Adams`,
  `Thomas`,
  `Anderson`,
  `Lewis`,
  `Jackson`,
  `Green`,
  `Wood`,
  `Parker`,
  `Campbell`,
  `Robinson`,
  `Scott`,
  `Phillips`,
  `Turner`,
  `Carter`,
];

export const features = [
  `Wi-fi`,
  `Dishwasher`,
  `Parking`,
  `Washer`,
  `Elevator`,
  `Conditioner`,
  `Gym`,
  `Guide`,
  `Video games`,
  `Towels`,
  `Kitchen`,
  `Heating`,
];

export const descriptions = [
  `Lorem ipsum dolor sit amet, consectetur adipiscingâ€¦empor incididunt ut labore et dolore magna aliqua`,
  `Odio aenean sed adipiscing diam donec adipiscing tristique risus nec`,
  `Ut ornare lectus sit amet est placerat`,
  `Habitant morbi tristique senectus et`,
  `Ac tortor vitae purus faucibus ornare suspendisse sed nisi lacus`,
  `Tortor aliquam nulla facilisi cras fermentum odio eu feugiat pretium`,
  `Mattis enim ut tellus elementum`,
  `Justo donec enim diam vulputate ut pharetra sit amet aliquam`,
  `Elit ut aliquam purus sit amet luctus venenatis lectus`,
  `Eget velit aliquet sagittis id`,
  `Mi sit amet mauris commodo quis imperdiet massa`,
  `Ullamcorper a lacus vestibulum sed arcu non odio`,
  `Elementum eu facilisis sed odio morbi quis commodo odio`,
  `Et malesuada fames ac turpis egestas integer eget aliquet nibh`,
  `Pharetra massa massa ultricies mi quis hendrerit dolor`,
  `Eu turpis egestas pretium aenean pharetra magna ac`,
  `Dictum fusce ut placerat orci`,
  `Nisi vitae suscipit tellus mauris a diam maecenas sed`,
  `Mauris sit amet massa vitae tortor condimentum lacinia quis vel`,
  `Id faucibus nisl tincidunt eget nullam non nisi est sit.`,
];

export const titles = [
  `Beautiful & luxurious apartment at great location`,
  `Couples particularly like this location `,
  `Big company apartment`,
  `Nearest airport & transport station`,
  `Quite place for zen vacations`,
  `Best lobby bar & Michelin restaurant`,
  `Bed and breakfast. Best offer!`,
];
