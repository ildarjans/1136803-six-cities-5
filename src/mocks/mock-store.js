import configureStore from "redux-mock-store";
import {
  mockCityOffers,
  mockFavoritesOffers,
  mockOffers,
  mockReviews,
  mockUser
} from "./mock-data";

const createStore = configureStore();

const mockStoreTemplate = {
  OFFERS: {
    offers: mockOffers,
    cityOffers: mockCityOffers,
    loading: true,
    error: null
  },
  REVIEWS: {
    reviews: mockReviews,
    loading: false,
    error: null,
    posting: false
  },
  NEAR_OFFERS: {
    nearOffers: mockCityOffers.AMSTERDAM.slice(0, 3),
    loading: false,
    error: null
  },
  PROCESS: {
    activeCity: `AMSTERDAM`,
    hoveredOfferId: ``,
    sortType: ``
  },
  USER: {
    authorizationStatus: `NO_AUTHORIZED`,
    waitingAuthorizationResponse: false,
    error: null,
    user: mockUser
  },
  FAVORITES: {
    favoriteCityOffers: mockFavoritesOffers,
    updating: false,
    loading: false,
    error: null,
  }
};

export const mockStore = createStore(mockStoreTemplate);
