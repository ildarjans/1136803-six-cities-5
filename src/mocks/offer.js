import {
  generateId,
  getLocationCoords,
  getRandomArrayElement,
  getRandomArrayElements,
  getRandomAvatar,
  getRandomFullName,
  getRandomImageGallery,
  getRandomInteger,
} from "./utils";
import {offerTypes, cities} from "../const";
import {
  descriptions,
  features,
  BEDROOMS_RANGE,
  GUEST_RANGE,
  MAX_FEATURES,
  PRICE_RANGE,
  titles,
} from "./consts";

export function getOffer() {
  const city = getRandomArrayElement(cities);
  return {
    city,
    id: generateId(),
    images: getRandomImageGallery(),
    premium: Boolean(getRandomInteger(1)),
    price: getRandomInteger(PRICE_RANGE.MAX, PRICE_RANGE.MIN),
    title: getRandomArrayElement(titles),
    type: getRandomArrayElement(offerTypes),
    rating: getRandomInteger(5),
    description: getRandomArrayElement(descriptions),
    bedrooms: getRandomInteger(BEDROOMS_RANGE.MAX, BEDROOMS_RANGE.MIN),
    guests: getRandomInteger(GUEST_RANGE.MAX, GUEST_RANGE.MIN),
    features: getRandomArrayElements(features, MAX_FEATURES),
    owner: {
      avatar: getRandomAvatar(),
      name: getRandomFullName(),
      super: Boolean(getRandomInteger(1)),
    },
    coords: getLocationCoords(city)
  };
}

