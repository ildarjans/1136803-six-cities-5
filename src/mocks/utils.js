import {
  AVATAR_URL,
  AVATAR_SPRITES,
  LAST_NAMES,
  NAMES,
  IMAGE_GALLERY,
} from "./consts";

export function getRandomInteger(max, min = 0) {
  return min + Math.floor(Math.random() * (max + 1 - min));
}

export function getRandomArrayElements(arr, count) {
  if (arr.length < count) {
    return undefined;
  }
  const resultArray = [];
  while (resultArray.length < count) {
    const randElement = arr[getRandomInteger(arr.length - 1)];
    if (!resultArray.includes(randElement)) {
      resultArray.push(randElement);
    }
  }
  return resultArray;
}

export function getRandomArrayElement(arr) {
  if (!arr && arr.length === 0) {
    return undefined;
  }
  return arr[getRandomInteger(arr.length - 1)];
}

export function getRandomFullName() {
  return (
    `${getRandomArrayElement(NAMES)} ${getRandomArrayElement(LAST_NAMES)}`
  );
}

export function getRandomAvatar() {
  return `${AVATAR_URL}/${getRandomArrayElement(AVATAR_SPRITES)}/${Math.random()}.svg`;
}

export function getRandomImageGallery() {
  const gallerySize = getRandomInteger(IMAGE_GALLERY.MAX, IMAGE_GALLERY.MIN);
  return Array(gallerySize).fill().map(getRandomPhoto);
}

function getRandomPhoto() {
  return `http://picsum.photos/248/152?r=${Math.random()}`;
}

export function generateId() {
  return getRandomInteger(Date.now()).toString(16);
}
