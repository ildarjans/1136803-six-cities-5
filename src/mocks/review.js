import {
  getRandomArrayElement,
  getRandomAvatar,
  getRandomFullName,
  getRandomInteger,
} from "./utils";
import {descriptions} from "./consts";

export function getReview() {
  return {
    avatar: getRandomAvatar(),
    name: getRandomFullName(),
    date: new Date(),
    rating: getRandomInteger(5),
    review: getRandomArrayElement(descriptions),
  };
}

