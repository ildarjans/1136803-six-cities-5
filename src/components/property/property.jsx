import React from "react";
import {withRouter} from "react-router-dom";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {getDecimalRating, getRatingWidth} from "../../utils";
import {offerPropTypes} from "../../prop-types/offer";
import {mapIconPropTypes} from "../../prop-types/map";
import {
  selectOffer,
  selectPropertyMapIcons,
  selectRouteId
} from "../../selectors/selectors";
import {processActionCreator} from "../../store/process/process-action";

import Map from "../map/map";
import {Header} from "../header/header";
import {PropertyGallery} from "../property-gallery/property-gallery";
import {NotFoundPage} from "../not-found-page/not-found-page";
import {PropertyReviews} from "../property-reviews/property-reviews";
import {NearPlaces} from "../near-places/near-places";
import {CityCoords, FAVORITE_BUTTON_OPTIONS} from "../../const";
import {FavoriteButton} from "../favorite-button/favorite-button";

export const PropertyComponent = ({offer, icons, id, setHoveredIcon}) => {
  if (!offer) {
    return <NotFoundPage/>;
  }

  React.useEffect(() => setHoveredIcon(+id), [+id]);

  const center = CityCoords[offer.city.name.toUpperCase()];

  return (
    <div className="page">

      <Header/>

      <main className="page__main page__main--property">
        <section className="property">
          <PropertyGallery offer={offer}/>
          <div className="property__container container">
            <div className="property__wrapper">

              {offer.isPremium &&
              (<div className="property__mark">
                <span>Premium</span>
              </div>)}

              <div className="property__name-wrapper">
                <h1 className="property__name">{offer.title}</h1>

                <FavoriteButton
                  options={FAVORITE_BUTTON_OPTIONS.PROPERTY}
                  id={id}
                  isFavorite={offer.isFavorite}
                />

              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${getRatingWidth(offer.rating)}%`}}/>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">
                  {getDecimalRating(offer.rating)}
                </span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {offer.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {offer.bedrooms}
                </li>
                <li className="property__feature property__feature--adults">
                  {offer.maxAdults}
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">

                  {offer.goods.map((good, i) => {
                    return (
                      <li className="property__inside-item"
                        key={`${good}-${i}`}>
                        {good}
                      </li>
                    );
                  })}

                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div
                    className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img className="property__avatar user__avatar"
                      src={offer.host.avatarUrl} width="74" height="74"
                      alt={`${offer.host.name} avatar`}/>
                  </div>
                  <span className="property__user-name">
                    {offer.host.name}
                  </span>
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {offer.description}
                  </p>
                </div>
              </div>

              <PropertyReviews id={id}/>

            </div>
          </div>
          <section className="property__map map">
            <Map center={center} icons={icons}/>
          </section>
        </section>

        <NearPlaces id={id}/>

      </main>
    </div>
  );
};

PropertyComponent.propTypes = {
  offer: offerPropTypes.isRequired,
  icons: PropTypes.arrayOf(mapIconPropTypes.isRequired).isRequired,
  id: PropTypes.number.isRequired,
  setHoveredIcon: PropTypes.func.isRequired,
};

const mapStateToProps = (state, props) => ({
  id: selectRouteId(props),
  offer: selectOffer(state, props),
  icons: selectPropertyMapIcons(state, props),
});

const mapDispatchToProps = (dispatch) => ({
  setHoveredIcon(id) {
    dispatch(processActionCreator.changeHoveredOfferId(id));
  }
});

export const Property = connect(mapStateToProps, mapDispatchToProps)(withRouter(PropertyComponent));
