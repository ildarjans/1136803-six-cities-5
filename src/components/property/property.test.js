import React from "react";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";

import {mockAmsterdamIcons, mockCityOffers} from "../../mocks/mock-data";
import {mockStore} from "../../mocks/mock-store";

import {Property} from "./property";

const noop = () => {};

it(`Should Property rendered correctly`, () => {
  const offer = mockCityOffers.AMSTERDAM[0];
  const tree = renderer.create(
      <Provider store={mockStore}>
        <MemoryRouter>
          <Property
            setHoveredIcon={noop}
            id={offer.id}
            icons={mockAmsterdamIcons}
            offer={offer}
          />
        </MemoryRouter>
      </Provider>,
      {
        createNodeMock: () => document.createElement(`div`)
      }
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
