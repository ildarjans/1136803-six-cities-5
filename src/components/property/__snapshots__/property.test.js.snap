// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Property rendered correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Big company apartment-0"
              className="property__image"
              src="http://picsum.photos/248/152?r=0.0922933754368811"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Big company apartment-1"
              className="property__image"
              src="http://picsum.photos/248/152?r=0.897019347231856"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Big company apartment
            </h1>
            <button
              className="property__bookmark-button  button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={31}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To
                 bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "20%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              1
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              9
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              2523
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Washer
                Heating
                Gym
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Christopher Jones avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/public/img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Christopher Jones
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Et malesuada fames ac turpis egestas integer eget aliquet nibh
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews ·
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Samuel Robinson avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://avatars.dicebear.com/api/female/222.svg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Samuel Robinson
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Elit ut aliquam purus sit amet luctus venenatis lectus
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-10-28"
                  >
                    October 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Samuel Robinson avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://avatars.dicebear.com/api/female/222.svg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Samuel Robinson
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Elit ut aliquam purus sit amet luctus venenatis lectus
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-11-28"
                  >
                    November 2020
                  </time>
                </div>
              </li>
            </ul>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/73"
                onClick={[Function]}
              >
                <img
                  alt="Big company apartment"
                  className="place-card__image"
                  height={200}
                  src="http://picsum.photos/248/152?r=0.16844568825898554"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    2523
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={18}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To
                     bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "20%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/73"
                  onClick={[Function]}
                >
                  Big company apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/85"
                onClick={[Function]}
              >
                <img
                  alt="Big company apartment"
                  className="place-card__image"
                  height={200}
                  src="http://picsum.photos/248/152?r=0.34977443093807237"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    3585
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={18}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In
                     bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/85"
                  onClick={[Function]}
                >
                  Big company apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
